components/Countdown.js

```diff
@@ -0,0 +1,39 @@
+var React = require("react");
+import PropTypes from "prop-types";
+import timediff from "timediff";
+class Countdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.tick = this.tick.bind(this);
+    this.state = { date: props.initialDate, now: Date.now() };
+    this.TimeDiff = this.timeDiff();
+  }
+
+  componentDidMount() {
+    setInterval(this.tick, 1000);
+  }
+  tick() {
+    this.setState({ now: Date.now() });
+    this.TimeDiff = this.timeDiff();
+  }
+  timeDiff() {
+    return timediff(this.state.now, this.state.date, "D:H:m:S");
+  }
+  render() {
+    return (
+      <div>
+        <p>
+          {this.TimeDiff.days} Days&#160;
```

Comment:
Storing deriving values is an anti-pattern. You could just call the `timediff` in the render method instead.